include/drivers/kinetis/i2c.h

```diff
@@ -0,0 +1,106 @@
+#pragma once
+#include <MK64F12.h>
+#include <stdbool.h>
+
+/*************
+ * CONSTANTS
+ *************/
+#define I2C_NUM_DEVICES 1
+
+#define I2C_AVAILABLE 0
+#define I2C_BUSY 1
+#define I2C_ERROR 2
+
+#define I2C_WRITING 0
+#define I2C_READING 1
+
+/**********
+ * MACROS
+ **********/
+#define I2C_WRITE_ADDR(x)   (i2c_seq_t){.opcode=I2C_WRITE, .addr=(x << 1)}
+#define I2C_READ_ADDR(x)    (i2c_seq_t){.opcode=I2C_WRITE, .addr=(x << 1) | 1}
+#define I2C_WRITE_REG(x)    (i2c_seq_t){.opcode=I2C_WRITE, .reg=x}
+#define I2C_WRITE_VALUE(x)  (i2c_seq_t){.opcode=I2C_WRITE, .value=x}
+#define I2C_SEND_READ       (i2c_seq_t){.opcode=I2C_READ}
+#define I2C_SEND_RESTART    (i2c_seq_t){.opcode=I2C_RESTART}
+
+#define I2C_TX_WAIT(x) while(i2c_chs[x].status != I2C_AVAILABLE);
+
+/**
+ * I2C Configuration struct.
+ */
+typedef struct {
+    uint8_t i2c_num;
+    uint32_t baud;
+    bool stophold;
+    bool interrupt;
+    uint8_t filter_clocks;  /* 0-15 for number of clocks */
+    uint8_t priority;
+} i2c_config_t;
+
+/**
+ * I2C Sequence struct. A sequence consists of an array of these structs.
+ */
+typedef struct {
+    enum opcode_t {
+        I2C_RESTART,
+        I2C_READ,
+        I2C_WRITE
+    } opcode;
+    union {
+        uint8_t addr;
+        uint8_t reg;
+        uint8_t value;
+    };
+} i2c_seq_t;
+
+/**
+ * I2C Channel state struct. Keeps track of the state of each channel.
+ */
+typedef struct {
+    i2c_seq_t *seq;
+    i2c_seq_t *seq_end;
+    uint8_t *received_data;
+    uint8_t reads_ahead;
+    uint8_t txrx;
+    uint8_t status;
+    void (*callback)(void *);
+    void *args;
+} I2C_Channel;
+
+/**
+ * Global variable that keeps track of the I2C Channel states.
+ */
+extern volatile I2C_Channel i2c_chs[I2C_NUM_DEVICES];
```

Comment:
Is there any need for this to be here? Generally speaking, modules external to this one won't need access to this variable directly, so there isn't any need for it to be visible outside this module.
Does the other code you intend to PR later require this? If not, we should remove this.

Reply:
This is required outside as the macro `I2C_TX_WAIT()` requires it to check the current status of the specific I2C channel. The wait macro is used in device initialization so that it blocks until a transfer is complete.

Reply:
Okay, seems good enough to me.  It's not worth coming up with a cleaner solution right now as this is plenty good enough.  That's user risk anyway.